### PR TITLE
Add a developer document describing the script repository

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

It describes each component and links to the production setup of the backend.

**To test:**

Review the wording.

*There is no associated issue.*

*This does not require release notes* because **it is internal documentation.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/ScriptRepository.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+.. _scriptrepository:
+
+=================
+Script Repository
+=================
+
+This page gives describes the implementation of the :ref:`script repository <WorkbenchScriptRepository>`
```
Comment (robertapplin) [inline dev-docs/source/ScriptRepository.rst:7]:
```suggestion
This page describes the implementation of the :ref:`script repository <WorkbenchScriptRepository>`
```

**Review by robertapplin — CHANGES_REQUESTED** [review]
Looks good, just spotted one small typo, then I'm happy to approve

File: dev-docs/source/ScriptRepository.rst (hunk 1 above)
Comment (martyngigg) [reply dev-docs/source/ScriptRepository.rst]: Nice catch! I've updated this now. Thanks.

**Review by robertapplin — APPROVED** [review]
Thank you, looks good!

**Pull request merged**